js.go

```diff
@@ -541,10 +541,14 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, opts []
 			deliver = NewInbox()
 		}
 	} else {
-		// Find the stream mapped to the subject.
-		stream, err = js.lookupStreamBySubject(subj)
-		if err != nil {
-			return nil, err
+		// Find the stream mapped to the subject if not bound to a stream already.
+		if o.stream == _EMPTY_ {
+			stream, err = js.lookupStreamBySubject(subj)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			stream = o.stream
```

Comment:
If we ask to bind to a stream this should be a hard failure if not in the list returned from the server, no?
Do I need to do any work on server side for matching? Right now I would imagine you will not see those streams in normal names lookup yes?

Reply:
Right now the [stream names API](https://github.com/nats-io/nats-server/blob/ee92cc9a5ba23a07f7678109ed5af3b36d38b393/server/jetstream_api.go#L1286) only returns the origin stream (does not include mirrors or stream with sources), so we avoid the lookup and just use what is presented in BindStream.
If the server is changed to also include all the mirrors/stream sources that match that subject, then in that case we would have to change the check here.

Reply:
ok so we need some changes on server side to return those streams as well when matching based on subject.

Reply:
Until then though I think this works, but let's revisit when names will return all streams etc..